components/protected-route/ProtectedRoute.tsx

```diff
@@ -0,0 +1,30 @@
+import { useAuthContext } from "@/context/AuthContext";
+import { useRouter } from "next/router";
+import { FC } from "react";
+
+
+
+const WithProtectedRoute = <P extends object>(
+    WrappedComponent: FC<P>
+): FC<P> => {
+    const Wrapper = (props: P) => {
+        const router = useRouter();
+        const { user, isFirstLogin } = useAuthContext();
+        
+        if (!user) {
+            router.push("./signin");
+        }
+
+        if (isFirstLogin) {
+            if(router.pathname !== "/change-password") {
+                router.push("/change-password");
+            }
+        }
+
+        return !user ? null : <WrappedComponent {...props} />;
+    };
+
+    return Wrapper;
```

Comment:
Is it necessary to create a local variable for this? Perhaps inline the variable.

Reply:
yes cuz hooks wont work without declaring first